src/scatter_nd.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+import {Tensor, sizeOfShape} from './lib/tensor.js';
+import {validateScatterNDParams} from './lib/validate-input.js';
+import {identity} from './unary.js';
+
+/**
+ * Scatter values using multidimensional indices.
+ * @param {Tensor} input
+ * @param {Tensor} indices
+ * @param {Tensor} updates
+ * @return {Tensor}
+ */
+export function scatterND(input, indices, updates) {
+  // Refer to https://onnx.ai/onnx/operators/onnx__ScatterND.html
+  // output = np.copy(data)
+  // update_indices = indices.shape[:-1]
+  // for idx in np.ndindex(update_indices):
+  //     output[indices[idx]] = updates[idx]
+
+  validateScatterNDParams(input, indices, updates);
+
+  const output = identity(input);
+  const inputRank = input.rank;
+  const inputShape = input.shape;
+  const indicesRank = indices.rank;
+  const indicesShape = indices.shape;
+  const indicesTotal = sizeOfShape(indicesShape);
+  const lastIndicesSize = indicesShape[indicesRank - 1];
+  const tmpShape = inputShape.slice(lastIndicesSize, inputRank);
```

Comment:
Using term "slice" seems more clear. Similar to comment to `gatherND`: https://github.com/webmachinelearning/webnn-baseline/pull/119#discussion_r1899358503

```suggestion
  const sliceShape = inputShape.slice(lastIndicesSize, inputRank);
```